src/async-wrap.cc

```diff
@@ -296,6 +296,14 @@ static void PromiseHook(PromiseHookType type, Local<Promise> promise,
   PromiseWrap* wrap = Unwrap<PromiseWrap>(promise);
   if (type == PromiseHookType::kInit || wrap == nullptr) {
     bool silent = type != PromiseHookType::kInit;
+    // set parent promise's async Id as this promise's triggerId
+    if (parent->IsPromise()) {
+      // parent promise exists, current promise
+      // is a chained promise, so we set parent promise's id as
+      // current promise's triggerId
+      double trigger_id = Unwrap<PromiseWrap>(parent.As<Promise>())->get_id();
```

Comment:
@JiaLiPassion I think you would still want to create a *new* `PromiseWrap` for the `parent` Promise. If you do that, you don’t need to check `if (parent_wrap != nullptr)`.

(Basically, look at the existing code here and how it adds a `parent_wrap` when there isn’t any. If it’s unclear, feel free to ping me and I’ll explain in more detail. :))